src/lib/gradientParser.test.ts

```diff
@@ -0,0 +1,173 @@
+import { assert, expect, test } from "vitest"
+import { parseGradient } from "./gradientParser"
+
+test("parseGradient(linear-gradient)", () => {
+  expect(parseGradient("linear-gradient(180deg, red)")).toStrictEqual(
+    parseGradient("linear-gradient(rgb(255, 0, 0))")
+  )
+  expect(parseGradient("linear-gradient(blue 0)")).toStrictEqual(
+    parseGradient("linear-gradient(blue 0px)")
+  )
+  expect(parseGradient("linear-gradient(to left, red, blue)")).toMatchInlineSnapshot(`
+    [
+      {
+        "colorStops": [
+          {
+            "rgba": {
+              "a": 1,
+              "b": 0,
+              "g": 0,
+              "r": 255,
+            },
+            "type": "color-stop",
+          },
+          {
+            "rgba": {
+              "a": 1,
+              "b": 255,
+              "g": 0,
+              "r": 0,
+            },
+            "type": "color-stop",
+          },
+        ],
+        "orientation": {
```

Comment:
I think if we follow the W3C Spec more closely with naming it will be far easier to reason with things.

This property is actually the "gradientLine" which can be either an `angle` or `sideOrCorner`. We can also camelCase them so they are easier to use in JS without doing `parsedGradient["gradient-line"]`. For an `or` it does make sense to have a `type` property there.

<img width="849" alt="CleanShot 2022-10-18 at 21 30 55@2x" src="https://user-images.githubusercontent.com/1386966/196598869-872c4454-0e4f-4b91-b9cd-7c189b469937.png">

What do you think?

Reply:
sounds good to me, I sticked with the naming from gradient-parser to avoid having an unreviewable PR, will do the rename in a separate commit on the PR so it's easier to review.

Reply:
done

Reply:
Thanks I'll take a look tomorrow, didn't get a chance to today